core/trino-main/src/main/java/io/trino/operator/ExchangeClient.java

```diff
@@ -275,34 +234,17 @@ public synchronized void close()
         for (HttpPageBufferClient client : allClients.values()) {
             closeQuietly(client);
         }
-        pageBuffer.clear();
+        buffer.close();
```

Comment:
`systemMemoryContext.setBytes(0);` below assumes that `buffer` does not consume any memory after it is closed. It kinda makes sense but may not be true always. Maybe use `systemMemoryContext.setBytes(buffer.getRetainedSizeInBytes());`

Reply:
The assumption is that the `ExchangeClient` will be released after close. From what I remember all the memory contexts have to be set to 0 before the task finishes, otherwise it may create an accounting leak here: https://github.com/trinodb/trino/blob/master/lib/trino-memory-context/src/main/java/io/trino/memory/context/SimpleLocalMemoryContext.java#L91